File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.java
public class Solution {

	public int[][] solution(int[][] arr1, int[][] arr2) {
		int width = arr2[0].length;
		int height = arr1.length;
		int[][] answer = new int[height][width];

		for (int i = 0; i < height; i++) {
			for (int j = 0; j < width; j++) {
				int value = 0;

				for (int k = 0; k < arr1[i].length; k++) {
					value += arr1[i][k] * arr2[k][j];
				}

				answer[i][j] = value;
			}
		}

		return answer;
	}
}

File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.py
from typing import List

def solution(arr1: List[List[int]], arr2: List[List[int]]) -> List[List[int]]:
    answer_width = len(arr2[0])
    answer_height = len(arr1)
    arr1_width = len(arr1[0])
    answer = [[] for _ in range(answer_height)]
    
    for i in range(answer_height):
        for j in range(answer_width):
            sums = 0
            for k in range(arr1_width):
                sums += arr1[i][k] * arr2[k][j]
            answer[i].append(sums)
        
    return answer